javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java

```diff
@@ -131,11 +132,12 @@ private static void installBytebuddyAgent(
         new AgentBuilder.Default(
                 // default method graph compiler inspects the class hierarchy, we don't need it, so
                 // we use a simpler and faster strategy instead
-                new ByteBuddy().with(MethodGraph.Compiler.ForDeclaredMethods.INSTANCE))
+                new ByteBuddy()
+                    .with(MethodGraph.Compiler.ForDeclaredMethods.INSTANCE)
+                    .with(VisibilityBridgeStrategy.Default.NEVER)
+                    .with(InstrumentedType.Factory.Default.FROZEN))
+            .with(AgentBuilder.TypeStrategy.Default.DECORATE)
```

Comment:
`DECORATE` javadocs say that:

> A decoration does not allow for any standard transformations but can be used as a performance optimization compared to a redefinition, especially when implementing a Java agent that only applies ASM-based code changes.

I'm not sure what "standard transformations" mean in this context; do you know if this has any extra limitations compared to the previous strategy?

Reply:
I had the same question, but the tests pass so it seems that it is working for our use case. I think that the advices that we use are only a small and very restrictive part of byte buddy transformations. It is possible that there is a difference when more complicated transforms are used.

Reply:
The only difference I noticed was the issue in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8391 where there was a failure instrumenting a static method with an advice that used `this`

Reply:
> but the tests pass

👍